### PR TITLE
CC: Remove target directory for umoci before unpacking

### LINKED ISSUE #5602: CC: kata-agent failed to unpack image with umoci
When using skopeo and umoci to download/unpack a container image, umoci complains that an existing config.json found in the target directory with the message:
```
 failed to unpack image: ExitStatus(unix_wait_status(256))
```

Note, this path is only activated when both `skopeo` and `umoci` are present in the guest image.

https://github.com/kata-containers/kata-containers/blob/38b5818813097f7e09de96798c4b319934ee9d49/src/agent/src/image_rpc.rs#L263

### PR DESCRIPTION
**Pull request by jxyang**
Fixes #5602

Signed-off-by: Xuejun Yang <xuejya@microsoft.com>

### CODE REVIEW

**Comment by fidencio** [comment]
@jxyang, thanks for the contribution.  We're currently in a freeze till `v0.2.0` is tagged, thus I'm adding the `do-not-merge` label.
This PR will be reviewed as soon as we tag a release on the Kata Containers side.

Thanks for the understanding.

**Comment by fidencio** [comment]
Removing the `do-not-merge` label as the Confidential Containers `v0.2.0` release is out.
Thanks a lot for your patience!

File: src/agent/src/image_rpc.rs [hunk 1]
```diff
@@ -129,6 +129,7 @@ impl ImageService {
         let source_path_oci = tmp_cid_path.join(IMAGE_OCI);
 
         let target_path_bundle = Path::new(CONTAINER_BASE).join(cid);
+        let _ = fs::remove_dir_all(&target_path_bundle);
```
Comment (fidencio) [inline src/agent/src/image_rpc.rs:132]: @stevenhorsman, mind to take a look?

It does make sense to me.

**Comment by stevenhorsman** [comment]
I'm not 100% sure, but it looks like the compilation errors are in some dragonball stuff and I'll note that with https://github.com/kata-containers/kata-containers/pull/5615 they updated/deleted a bunch of dependencies recently, so I wonder if we need to do a merge from `main` soon to get these fixes in?

**Comment by stevenhorsman** [comment]
> I'm not 100% sure, but it looks like the compilation errors are in some dragonball stuff and I'll note that with #5615 they updated/deleted a bunch of dependencies recently, so I wonder if we need to do a merge from `main` soon to get these fixes in?

https://katacontainers.slack.com/archives/C01HUGDR0CE/p1668138475408719 suggested this might be the case. I'm not sure if the fix has gone in yet though, so we might need to wait before merging

**Comment by jxyang** [comment]
Thanks for the heads up. @stevenhorsman. Yes we plan to migrate away from Skopeo in the future. But for now we still need this fix.

I seem having problem opening the slack discussion. If the underlying issue fixed?

**Comment by stevenhorsman** [comment]
> Thanks for the heads up. @stevenhorsman. Yes we plan to migrate away from Skopeo in the future. But for now we still need this fix.
> 
> I seem having problem opening the slack discussion. Is the underlying issue fixed? And is there a way for me to trigger the rerun of the pipeline?

I think the PR we need fixed upstream is https://github.com/kata-containers/kata-containers/pull/5658. Then after that we need to merge the `main` content into `CCv0` and if you rebase on the latest `CCv0` then the static checks should be fixed. I'll try and keep you updated on the status as we move through this process and sorry about the delays.

**Comment by stevenhorsman** [comment]
@jxyang - sorry about the delays with had with upstream changes in the CI process. We think we have all the fixes merged, so can you rebase this PR and then we'll get the tests running. Thanks

**Comment by stevenhorsman** [comment]
Given we've removed the umoci support from the agent, I think this PR isn't valid, so should be closed.

**Pull request closed without merging**